lib/pbench/server/api/resources/__init__.py

```diff
@@ -0,0 +1,686 @@
+        authorization header, the requested user, the requested access
+        policy, and the API's role.
+
+        If "user" is None, then the request is unauthenticated. READ operations
+        may be allowed, UPDATE and DELETE operations will not be allowed.
+
+        If "access" is None, READ will assume we're looking for access only to
+        public datasets.
+
+        for API_OPERATION.READ:
+
+            Any call, with or without an authenticated user token, can access
+            public data.
+
+            Any authenticated user can access their own private data.
+
+            Any authenticated ADMIN user can access any private data.
+
+        for API_OPERATION.UPDATE:
+
+            An authenticated user is required.
+
+            Any authenticated user can update their own data.
+
+            Any authenticated ADMIN user can update any data.
+
+        Args:
+            user: The user parameter to the API, or None
+            access: The access parameter to the API, or None
+
+        Raises:
+            UnauthorizedAccess The user isn't authorized for the requested
+                access.
+        """
+        authorized_user: User = Auth.token_auth.current_user()
+        authorized = True
+        self.logger.debug(
+            "Authorizing {} access for {} to user {} with access {}",
+            self.role,
+            authorized_user,
+            user,
+            access,
+        )
+        if self.role != API_OPERATION.READ or access == Dataset.PRIVATE_ACCESS:
+            if authorized_user is None:
+                self.logger.warning(
+                    "Attempt to {} user {} data without login", self.role, user
+                )
+                authorized = False
+            elif user != authorized_user.username and not authorized_user.is_admin():
+                self.logger.warning(
+                    "Unauthorized attempt by {} to {} user {} data",
+                    authorized_user,
+                    self.role,
+                    user,
+                )
+                authorized = False
+        if not authorized:
+            raise UnauthorizedAccess(authorized_user, self.role, user, access)
+
+    def _dispatch(self, method: Callable, request: Request) -> Response:
+        """
+        This is a common front end for HTTP operations, which will validate and
+        normalize the request JSON document if there's a class schema and the
+        operation isn't GET (which doesn't support a request payload).
+
+        Args:
+            method: A reference to the implementation method
+            request: The flask Request object containing payload and headers
+
+        Returns:
+            Flask Response object generally constructed implicitly from a JSON
+            payload and HTTP status.
+        """
+        json_data = request.get_json(silent=True)
+        if self.schema and method != self._get:
```

Comment:
The docstring implies that this method won't be called if the `method` is for `GET`, so it's slightly weird to be testing for `_get()`...is the docstring misworded, or is this just belt-and-suspenders?

Also, I think it would yield nicer code if you inverted this test and returned immediately if it _is_ a `GET` -- that way the rest of the function body doesn't need the extra level of indentation.

Reply:
The docstring says that it will "validate and normalize [if] the operation isn't a GET". Apparently the language wasn't sufficiently clear.